docs/manual/working/scalaGuide/main/json/ScalaJsonAutomated.md

```diff
@@ -34,13 +34,15 @@ The above example can be made even more concise by using body parsers with a typ
 
 ### Requirements
 
-These macros rely on a few assumptions about the type they're working with :
-- It must have a companion object having `apply` and `unapply` methods
+These macros support classes and sealed traits. It relies on the following assumptions.
+
+- It must have a companion object having `apply` and `unapply` methods.
 - The return types of the `unapply` must match the argument types of the `apply` method.
 - The parameter names of the `apply` method must be the same as the property names desired in the JSON.
 
-Case classes natively meet these requirements. For more custom classes or traits, you might
-have to implement them.
+Case classes automatically meet these requirements. For custom classes or traits, you might have to implement them.
+
+A trait can also supported, if and only if it's a sealed one and if the sub-types comply with the previous requirements.
```

Comment:
The wording is a bit strange. How about "A sealed trait can be supported if all the subtypes comply with the above requirements". Then it would be nice to explain how sealed trait instances are represented in JSON.

Reply:
It would no longer emphasise that only sealed trait is supported, precision that seems important to me

Reply:
Aren't unsealed traits supported if they have matching apply/unapply in their companion?

Reply:
That's the previous case for me ("For custom classes or traits, you might have to implement them").

There is more about sealed family.

Reply:
That's why I thought saying "A sealed trait is supported if ..." is more clear than "A trait can be supported if and only if ...".

Reply:
I find clearer to indicate that `trait` if complying with the 2 conditions (being sealed, and compliant sub-types), rather than "jumping" directly to `sealed trait`.

I mean, being `sealed` and having compliant sub-types are conditions of the same importance, to be considered for the `trait` concept.